homeassistant/components/litterrobot/config_flow.py

```diff
@@ -36,22 +68,26 @@ async def async_step_user(
         if user_input is not None:
             self._async_abort_entries_match({CONF_USERNAME: user_input[CONF_USERNAME]})
 
-            hub = LitterRobotHub(self.hass, user_input)
-            try:
-                await hub.login()
-            except LitterRobotLoginException:
-                errors["base"] = "invalid_auth"
-            except LitterRobotException:
-                errors["base"] = "cannot_connect"
-            except Exception:  # pylint: disable=broad-except
-                _LOGGER.exception("Unexpected exception")
-                errors["base"] = "unknown"
-
-            if not errors:
+            if not (error := await self._async_validate_input(user_input)):
                 return self.async_create_entry(
                     title=user_input[CONF_USERNAME], data=user_input
                 )
+            errors["base"] = error
 
         return self.async_show_form(
             step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
         )
+
+    async def _async_validate_input(self, user_input: Mapping[str, Any]) -> str:
+        """Validate login credentials."""
+        hub = LitterRobotHub(self.hass, user_input)
+        try:
+            await hub.login()
+        except ConfigEntryAuthFailed:
```

Comment:
We shouldn't raise ConfigEntry exceptions in a config flow. These are only intended to be raised during setup of a config entry.